### PR TITLE
Backport standard Kubernetes clients for AuthConfigs to 1.4

### PR DESCRIPTION
**Pull request by marcogschmidt**
Backport of https://github.com/solo-io/gloo/pull/3644
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/3643

### CODE REVIEW

File: changelog/v1.4.14/native-authconfig-type.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+changelog:
+- type: FIX
```
Comment (kdorosh) [inline changelog/v1.4.14/native-authconfig-type.yaml:2]: non user facing? thoughts?

Comment (marcogschmidt) [reply changelog/v1.4.14/native-authconfig-type.yaml:2]: Not sure to be honest, the intention was to make it user facing but then again calling this a fix might be a stretch.

**Review by kdorosh — DISMISSED** [review]

**Review by kdorosh — APPROVED** [review]

**Pull request merged**